src/main.c

```diff
@@ -288,31 +291,67 @@ static uint32_t read_encoder_right(void)
 
 
 /**
+ * @brief Setup for adc: Four injected channels on ADC1.
+ */
+static void setup_adc(void)
+{
+	int i;
+	uint8_t channel_array[4];
```

Comment:
I would rather put here the initialization of the channel array. In example:

    uint8_t channel_sequence[] = [ADC_CHANNELX, ADC_CHANNELY, ... ADC_CHANNELZ];

Pseudocode... :joy:

Reply:
parenthesis is the key